### ISSUE TITLE
MSRV is 1.71 and not 1.70

### ISSUE THREAD

**Issue by pksunkara** [obi1kenobi/cargo-semver-checks#597]
### Steps to reproduce the bug with the above code

Since I use the code as a subdmodule in a different project, I can't write the reproduction steps here. But I assume, running the CLI would be enough.

All I can say is that local rust version is 1.70

### Actual Behaviour

```
rustdoc format v24 for file ....target/semver-checks/target/doc/clap.json is not supported
```

### Expected Behaviour

Tool to work

### Generated System Information

Used the code from the commit 438f6d80a368e9d119159ce44e27ca1bf692b9ab which is a few days ago.

**Comment by obi1kenobi** [comment]
Yes, you're totally right — sorry for finding this out the hard way. I accidentally failed to include this in the release notes for v0.25. I have now updated the release notes.

Rustdoc prior to Rust 1.73 has a bug where re-exports of `#[doc(hidden)]` items are incorrectly omitted from the JSON output, which can cause false-positive lints to be reported. Rust 1.71-1.73 all use the same rustdoc JSON format so technically the MSRV is 1.71, but 1.73+ is strongly recommended. I dropped Rust 1.70 which used an older rustdoc JSON format, and would have dropped 1.71-1.72 as well if that were feasible, but it was not.

Unfortunately I failed to include this in the release notes, and instead it remained in my draft release. Sorry about that. I've since updated the release.

(Sidenote: `cargo-semver-checks` has two different flavors of MSRV: the minimum version which it requires to compile, and the minimum version whose rustdoc it can check. This affects only the latter (can't check Rust 1.70), not the former (it can still compile and work fine on 1.70, as you found). The "compile MSRV" should always be <= the "check MSRV.")

**Comment by obi1kenobi** [comment]
Thanks for highlighting these. I should add more info to the README.

> 1. There is a reference to 1.70.0 in `README.md` and `Cargo.toml`. I assume that you want them to also represent the check MSRV.

Those are both the compile MSRV, since `Cargo.toml` is what `cargo install` will read.

> 2. I don't think there's a need to ever talk about compile MSRV at all, because we can always treat the check MSRV as the only important one.

This is tricky. In principle, it should be true. In practice, we've had (rare!) situations where the compile MSRV was actually higher than the check MSRV since our dependencies' MSRV got bumped up but the rustdoc JSON format remained the same.

Rust 1.70 was such a case: lots of libraries bumped to MSRV 1.70 but rustdoc v24 was in 1.68, 1.69, and 1.70 — so cargo-semver-checks needed to be compiled on 1.70 but could then check output as old as 1.68. This is the only time in ~a year and a half this has happened, but I think it could happen again in the future so I'm hesitant to make the change you suggest.

**Comment by obi1kenobi** [comment]
Opened #598 to add more detail in the MSRV section of the README. If you have a chance to check it out, I'd love your input (but no pressure).
